test/integration/consul-container/test/upgrade/l7_traffic_management/traffic_management_test.go

```diff
@@ -0,0 +1,140 @@
+package l7trafficmanagement
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/consul/api"
+	libassert "github.com/hashicorp/consul/test/integration/consul-container/libs/assert"
+	libcluster "github.com/hashicorp/consul/test/integration/consul-container/libs/cluster"
+	libservice "github.com/hashicorp/consul/test/integration/consul-container/libs/service"
+	"github.com/hashicorp/consul/test/integration/consul-container/libs/topology"
+	"github.com/hashicorp/consul/test/integration/consul-container/libs/utils"
+	"github.com/stretchr/testify/require"
+)
+
+// TestTrafficManagement_Upgrade Summary
+// This test starts up 3 servers and 1 client in the same datacenter.
+//
+// Steps:
+//   - Create a single agent cluster.
+//   - Create one static-server and 2 subsets and 1 client and sidecar, then register them with Consul
+//   - Validate static-server and 2 subsets are and proxy admin endpoint is healthy
+//   - Validate static servers proxy listeners should be up and have right certs
+func TestTrafficManagement_SetupServerAndClientWithSubsets(t *testing.T) {
+	t.Parallel()
+
+	type testcase struct {
+		oldversion    string
+		targetVersion string
+	}
+	tcs := []testcase{
+		{
+			oldversion:    "1.14",
+			targetVersion: utils.TargetVersion,
+		},
+	}
+
+	run := func(t *testing.T, tc testcase) {
+		cluster, _, _ := topology.NewPeeringCluster(t, "dc1", 1, tc.oldversion)
+
+		// Register service resolver
+		serviceResolver := &api.ServiceResolverConfigEntry{
+			Kind:          api.ServiceResolver,
+			Name:          libservice.StaticServerServiceName,
+			DefaultSubset: "v2",
+			Subsets: map[string]api.ServiceResolverSubset{
+				"v1": {
+					Filter: "Service.Meta.version == v1",
+				},
+				"v2": {
+					Filter: "Service.Meta.version == v2",
+				},
+			},
+		}
+		err := cluster.ConfigEntryWrite(serviceResolver)
+		require.NoError(t, err)
+
+		serverService, serverServiceV1, serverServiceV2, clientService := createService(t, cluster)
+
+		_, port := clientService.GetAddr()
+		_, adminPort := clientService.GetAdminAddr()
+		_, serverAdminPort := serverService.GetAdminAddr()
+		_, serverAdminPortV1 := serverServiceV1.GetAdminAddr()
+		_, serverAdminPortV2 := serverServiceV2.GetAdminAddr()
+
+		// validate client and proxy is up and running
+		libassert.AssertContainerState(t, clientService, "running")
+		libassert.HTTPServiceEchoes(t, "localhost", port, "")
```

Comment:
I think we are missing the step of verifying the response is from s2-v2, which depends on Nick's PR. So could you add a TODO above line 69.